### ISSUE TITLE
Checks using match=search  or "match=regex not working in 2.2.2, 2.2.3

### ISSUE THREAD

**Issue by benjaminssmith** [NagiosEnterprises/ncpa#742]
Checks using  match=search will fail. For example, this command does not work in 2.2.2 and 2.2.3
```
/check_ncpa.py -H 192.168.xx.xx -t 'mytoken' -M 'services' -q 'service=ncpa,status=running,match=search'
UNKNOWN: An error occured connecting to API. (HTTP error: '500 INTERNAL SERVER ERROR'
```
This check syntax was previously working in version 2.2.0.

Log error
```
2021-02-05 12:47:46,933:INFO:handler:128.208.169.18 - - [2021-02-05 12:47:46] "GET /api/services?service=Sophos&match=search HTTP/1.1" 200 630 0.149000
2021-02-05 12:55:01,197:ERROR:app:Exception on /api/services [GET]
Traceback (most recent call last):
File "C:\Python27\lib\site-packages\flask\app.py", line 1817, in wsgi_app
File "C:\Python27\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
File "C:\Python27\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
File "C:\Python27\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
File "C:\Python27\lib\site-packages\flask\app.py", line 1461, in dispatch_request
File "C:\ncpa\agent\listener\server.py", line 292, in token_auth_decoration
File "C:\ncpa\agent\listener\server.py", line 1099, in api
File "C:\ncpa\agent\listener\services.py", line 365, in run_check
ValueError: 'Sophos System Protection Service' is not in list
```

**Customer Forum Post Submitted**
https://support.nagios.com/forum/viewtopic.php?t=61556#325073

**Comment by jomann09** [comment]
After looking into this a bit, I think it's the same issue as #626 which I thought was fixed in 2.3 because I was testing it wrong. I will be fixing this soon and get a 2.3.1 out with the fix.

**Comment by jomann09** [comment]
I've fixed this in 3d17056639e13546634c592eb86f421efc39a8ee and plan on releasing 2.3.1 next week.
